src/legacy/ui/public/agg_types/controls/top_aggregate.tsx

```diff
@@ -104,7 +98,7 @@ function TopAggregateParamEditor({
         setValue(aggParam.options.byValue[filteredOptions[0].value]);
       }
     },
-    [fieldType, visName]
+    [filteredOptions]
```

Comment:
Using `filteredOptions` to watch will evoke the effect on every render. Is it necessary?

Reply:
Maybe not, you know that code better, what could we actually use here?

Reply:
Do we have a way to change the vis type on the fly? I mean save the agg config and use the `gauge` instead of `area` ? If not, I think we could just to watch `fieldType` changes

Reply:
That's correct we can actually not really change that type, let me fix that.